### PR TITLE
Nimbus backend initial implementation

### PR DESCRIPTION
**Pull request by pedropombeiro**
The intent of this PR is to merge the Nimbus work done so far in a way that doesn't impact the way we're using status-go (with geth).

Changes:
  - Replaced `mobile/backend.go` with a pared-down Nimbus version (non-implemented methods will panic for the time being);
  - Still using a couple of Geth services instead of Nimbus replacements (waiting on @kdeme here):
    - keystore while we work to expose the Nimbus version;
    - RPC client/server classes;
  - Tests have not been adapted for Nimbus;
  - Some services have been disabled for Nimbus (mailservers, wallet, personal, discovery);

Thought process/architecture behind the change;
  - status-react will leverage Nix to build a `libnimbus.a` lib and feed it to status-go;
  - The idea is to fork logic at the backend level with the entry point used by status-react (`mobile/backend.go`);
  - We'll do compile-time backend selection (controlled by feature flag in `.env` files). Runtime selection is possible given some time investment;
  - Make it possible to iterate with local repos without needing to commit Nimbus/status-go changes;

Part of https://github.com/status-im/status-go/issues/1677

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -282,7 +293,8 @@ ci: lint canary-test test-unit test-e2e ##@tests Run all linters and tests at on
 ci-race: lint canary-test test-unit test-e2e-race ##@tests Run all linters and tests at once + race
 
 clean: ##@other Cleanup
-	rm -fr build/bin/* mailserver-config.json
+	rm -fr build/bin/* mailserver-config.json vendor/github.com/status-im/nimbus
```
Comment (jakubgs) [inline Makefile:294]: wat?

File: account/accounts_nimbus.go [hunk 2]
```diff
@@ -6,23 +6,24 @@ import (
...
+	m.Manager = &Manager{accountsGenerator: generator.New(m)}
 	return m
 }
 
-// InitKeystore sets key manager and key store.
-func (m *Manager) InitKeystore(keydir string) error {
-	m.mu.Lock()
-	defer m.mu.Unlock()
+// // InitKeystore sets key manager and key store.
+// func (m *Manager) InitKeystore(keydir string) error {
```
Comment (jakubgs) [inline account/accounts_nimbus.go:22]: Please don't commit commented out stuff like this.

File: account/keystore_nimbus.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+
+	gethbridge "github.com/status-im/status-go/eth-node/bridge/geth"
+	// nimbusbridge "github.com/status-im/status-go/eth-node/bridge/nimbus"
+	"github.com/status-im/status-go/eth-node/types"
+)
+
+// makeAccountManager creates ethereum accounts.Manager with single disk backend and lightweight kdf.
+// If keydir is empty new temporary directory with go-ethereum-keystore will be intialized.
+func makeAccountManager(keydir string) (manager *accounts.Manager, err error) {
```
Comment (jakubgs) [inline account/keystore_nimbus.go:19]: We're doing a LOT of code duplication with stuff like this. The only difference I see here is the name of the temporary directory which is `nimbus-keystore`. Do we really have to duplicate all this code just to change that?

I guess the main difference is the `gethbridge` used. If that's the case then why can't we parametrize it to accept an interface that can accept both types of bridges?

File: api/nimbus_backend.go [hunk 4]
```diff
@@ -0,0 +1,1027 @@
...
+	defer log.Info("Status backend initialized", "backend", "nimbus", "version", params.Version, "commit", params.GitCommit)
+
+	statusNode := node.NewNimbus()
+	accountManager := account.NewGethManager()
+	// transactor := transactions.NewTransactor()
+	// personalAPI := personal.NewAPI()
+	// rpcFilters := rpcfilters.New(statusNode)
+	return &nimbusStatusBackend{
+		statusNode:     statusNode,
+		accountManager: accountManager,
```
Comment (jakubgs) [inline api/nimbus_backend.go:90]: Same here. All this code duplication just because we want to use a different `AccountManager`? Is that really wise?

**Review by jakubgs — COMMENTED** [review]
I think a call to explain this PR would be cool.

File: account/accounts_nimbus.go (hunk 2 above)
Comment (pedropombeiro) [reply account/accounts_nimbus.go:22]: This is kind of a special case, where we're committing a partially implemented feature that's hidden behind a feature flag. I'm happy to discuss it on a call. Maybe you'd like to join the handover call that I'll be having with @andremedeiros?

**Review by jakubgs — APPROVED** [review]
Approving based on conversation I had with Pedro on Thursday. I understand that many of those changes are in a sense temporary and it is intentional that we are avoiding refactoring in order to simplify reaching functionality parity step by step.

**Pull request merged**